Fix explicit FTP adding 'ftps' as protocol and added control of implicit/explicit

@package vladimir-yuldashev/flysystem-curlftp

--- src/CurlFtpAdapter.php
+++ src/CurlFtpAdapter.php
@@ -13,6 +13,9 @@

 class CurlFtpAdapter extends AbstractFtpAdapter
 {
+    const SSL_IMPLICIT = 'implicit';
+    const SSL_EXPLICIT = 'explicit';
+
     protected $configurable = [
         'host',
         'port',
@@ -61,12 +64,18 @@
     protected $proxyPassword;

     /**
-     * @param bool $ssl
+     * @param bool|string $ssl
      */
     public function setSsl($ssl)
     {
-        $this->ssl = (bool) $ssl;
-    }
+        if (!$ssl) {
+            $ssl = false;
+        } else if (is_bool($ssl)) {
+            $ssl = static::SSL_IMPLICIT;
+        }
+
+        $this->ssl = $ssl;
+     }

     /**
      * @param int $sslVerifyPeer
@@ -697,7 +706,7 @@
      */
     protected function getBaseUri()
     {
-        $protocol = $this->ssl ? 'ftps' : 'ftp';
+        $protocol = $this->ssl === static::SSL_IMPLICIT ? 'ftps' : 'ftp';

         return $protocol.'://'.$this->getHost().':'.$this->getPort();
     }